include/deal.II/distributed/fully_distributed_tria.h

```diff
@@ -0,0 +1,406 @@
+#ifndef DOXYGEN
+namespace internal
+{
+  namespace DoFHandlerImplementation
+  {
+    namespace Policy
+    {
+      template <typename>
+      class ParallelDistributed;
+    } // namespace Policy
+  }   // namespace DoFHandlerImplementation
+} // namespace internal
+
+namespace FETools
+{
+  namespace internal
+  {
+    template <int, int, class>
+    class ExtrapolateImplementation;
+  }
+} // namespace FETools
+
+// forward declaration of the data type for periodic face pairs
+namespace GridTools
+{
+  template <typename CellIterator>
+  struct PeriodicFacePair;
+}
+#endif
+
+namespace parallel
+{
+  /**
+   * A namespace for the fully distributed Triangulation.
+   *
+   * @ingroup parallel
+   */
+  namespace fullydistributed
+  {
+    /**
+     * Information needed for each locally relevant cell, stored in
+     * ConstructionData and used during construction of a
+     * parallel::fullydistributed::Triangulation. This struct stores
+     * the cell id, the subdomain_id and the level_subdomain_id as well as
+     * information related to manifold_id and boundary_id.
+     *
+     * @author Peter Munch, 2019
+     */
+    template <int dim>
+    struct CellInfo
+    {
+      /**
+       * Constructor.
+       */
+      CellInfo() = default;
+
+      /**
+       * Unique CellID of the cell.
+       */
+      CellId::binary_type id;
+
+      /**
+       * subdomain_id of the cell.
+       */
+      types::subdomain_id subdomain_id;
+
+      /**
+       * level_subdomain_id of the cell.
+       */
+      types::subdomain_id level_subdomain_id;
+
+      /**
+       * Manifold id of the cell.
+       */
+      types::material_id manifold_id;
+
+      /**
+       * Manifold id of all lines of the cell.
+       *
+       * @note Only used for 2D and 3D.
+       */
+      std::array<types::material_id, GeometryInfo<dim>::lines_per_cell>
+        manifold_line_ids;
+
+      /**
+       * Manifold id of all faces of the cell.
+       *
+       * @note Only used for 3D.
+       */
+      std::array<types::material_id, GeometryInfo<dim>::quads_per_cell>
+        manifold_quad_ids;
+
+      /**
+       * List of face number and boundary id of all non-internal faces of the
+       * cell.
+       */
+      std::vector<std::pair<unsigned int, types::boundary_id>> boundary_ids;
+    };
+
+
+
+    /**
+     * Data used to construct a parallel::fullydistributed::Triangulation.
+     *
+     * @author Peter Munch, 2019
+     */
+    template <int dim, int spacedim>
+    struct ConstructionData
+    {
+      /**
+       * Cells of the locally-relevant coarse-grid triangulation.
+       */
+      std::vector<CellData<dim>> coarse_cells;
+
+      /**
+       * Vertices of the locally-relevant coarse-grid triangulation.
+       */
+      std::vector<Point<spacedim>> coarse_cell_vertices;
+
+      /**
+       * Mapping from coarse-grid index to coarse-grid id.
+       */
+      std::vector<types::coarse_cell_id> coarse_cell_index_to_coarse_cell_id;
```

Comment:
Can you elaborate what the two are? Maybe just be referencing the glossary entry I added a while ago.

Reply:
Also, it's a mapping, but a particular one: It's really just a list that, for each locally owned (relevant?) coarse mesh cell provides the corresponding global coarse-grid id. I think that a sentence like that might be easier to read in the description of the field.